### ISSUE TITLE
VC: requests may not contain Authorization header with API token

### ISSUE THREAD

**Issue by dnkolegov** [sigp/lighthouse#2512]
## Description

According to the [Validator Client API documentation](https://lighthouse-book.sigmaprime.io/api-vc-endpoints.html#post-lighthousevalidators), all requests (GET, POST, PATCH) must contain Authorization header with the token.

The current implementation only requires the API token in GET requests only, POST/PATCH requests will be processed even if they don't contain the Authorization header.

https://github.com/sigp/lighthouse/blob/stable/validator_client/src/http_api/mod.rs#L480

![image](https://user-images.githubusercontent.com/6521369/129372547-baf6cee8-fbc5-4524-a88c-c02ab1fbbcbd.png)

cc: @Nocasis

**Comment by paulhauner** [comment]
Thank you for raising this @dnkolegov :pray: It is indeed an issue and I have a fix over at #2517. We will include this in our next release.

In my testing I noticed that it's only `POST` requests that are missing header checks, whilst both `GET` and `PATCH` are checked. This is at odds with your description; you stated that `PATCH` was also passing without header checks.

The only reason I'm raising this is to make sure that I didn't miss something with the `PATCH` requests.

### Examples

PATCH indicates a missing header:

```
curl -X PATCH localhost:5062/lighthouse/validators
{"code":400,"message":"BAD_REQUEST: missing Authorization header","stacktraces":[]}
```

POST indicates an invalid body (meaning it didn't check the headers):

```
curl -X POST localhost:5062/lighthouse/validators
{"code":400,"message":"BAD_REQUEST: body deserialize error: Request body deserialize error: EOF while parsing a value at line 1 column 0","stacktraces":[]}
```

**Comment by paulhauner** [comment]
Resolved in #2517

**Comment by dnkolegov** [comment]
Hey @paulhauner.

```
curl -X PATCH localhost:5062/lighthouse/validators
{"code":400,"message":"BAD_REQUEST: missing Authorization header","stacktraces":[]}
```
Your example is incorrect. There is no such endpoint in the API.

I attached two requests illustrating that PATCH requests don't require tokens.
Please fix.

![photo_2021-08-18_10-06-52](https://user-images.githubusercontent.com/6521369/129830640-d2ddba58-e229-4cc8-aa6c-33f5d4f3c408.jpg)

![photo_2021-08-18_10-20-54](https://user-images.githubusercontent.com/6521369/129831755-c5685415-e0a1-4d0a-8472-6a5789d763e0.jpg)

**Comment by michaelsproul** [comment]
@dnkolegov You're right, PATCH requests were also bypassing the header check. Both POST and PATCH are fixed and tested as of #2517.

The expected behaviour is a 400 error with the message about the Authorization header (your first screenshot).
